internal/ingress/controller/nginx.go

```diff
@@ -628,6 +628,16 @@ func (n *NGINXController) OnUpdate(ingressCfg ingress.Configuration) error {
 		ListenPorts:                 n.cfg.ListenPorts,
 		PublishService:              n.GetPublishService(),
 		DynamicConfigurationEnabled: n.cfg.DynamicConfigurationEnabled,
+		LuaRestyWAFEnabled: func() bool {
```

Comment:
I prefer a setting in the configuration configmap to enable/disable the feature like [this](https://github.com/kubernetes/ingress-nginx/blob/master/docs/user-guide/configmap.md#enable-modsecurity)

Reply:
This is merely for configuring the shared dictionary if there's an ingress that enabled WAF: https://github.com/kubernetes/ingress-nginx/pull/2304/files#diff-980db9e4b88704f12338bd074839f94eR49

The feature is enabled per ingress.

But I acknowledge this is not the best place for that flag - I'm gonna move it to `template.go`.

Reply:
@aledbf just to make sure we are on the same page: are you suggesting instead of letting users to enable the feature per ingress we should do it globally using configmap(just like modsecurity)?

Reply:
> are you suggesting instead of letting users to enable the feature per ingress

This is fine. We just need a way to disable the feature in one place (configuration configmap)

Reply:
alright then, I'll introduce a new configmap setting `disable-lua-resty-waf`.